### PR TITLE
lesson 5 homework

### PR DESCRIPTION
**Pull request by jaredmulholland**

### CODE REVIEW

File: students/jared_mulholland/lesson_5/mail_room.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+main_prompt = ("\nMain Menu  \n 1. Send a Thank You \n 2. Create a Report \n 3. Send Letters \n 4. Quit \n Please Choose an Option: ")
+
+def mail_room_fun(main_prompt, main_dict):
+    while True:        
+        response = input(main_prompt)
+        if response == "4":
+            break
+        else:
+            print("\n")
+            main_dict.get(response)()
```
Comment (natasha-aleksandrova) [inline students/jared_mulholland/lesson_5/mail_room.py]: should have an exception handling here for invalid entry (ideally KeyError)

File: students/jared_mulholland/lesson_5/mail_room.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+"""
+CREATE A REPORT
+If the user (you) selected “Create a Report”, print a list of your donors, sorted by total historical donation amount.
+Include Donor Name, total donated, number of donations and average donation amount as values in each row. You do not need to print out all their donations, just the summary info.
+Using string formatting, format the output rows as nicely as possible. The end result should be tabular (values in each column should align with those above and below)
+After printing this report, return to the original prompt.
+At any point, the user should be able to quit their current task and return to the original prompt.
+From the original prompt, the user should be able to quit the script cleanly.
+"""
+import numpy as np
```
Comment (natasha-aleksandrova) [inline students/jared_mulholland/lesson_5/mail_room.py]: imports go at the top of the file

File: students/jared_mulholland/lesson_5/mail_room.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    
+    try:
+        thankyou = f"""\nDear {name.split()[0]}, \n\nThank you for your generous donation of ${new_donation:,.2f}. Thanks to you we will finally be able to begin construction 
+on the {name.split()[1]} Memorial Children's wing at Dark Place Hospital\n\nYours Truly,\nJared Mulholland"""
+    except IndexError:
+        thankyou = f"""\nDear {name.split()[0]}, \n\nThank you for your generous donation of ${new_donation:,.2f}. Thanks to you we will finally be able to begin construction 
+on the {name.split()[0]} Memorial Children's wing at Dark Place Hospital\n\nYours Truly,\nJared Mulholland"""
+
+    print(thankyou)
+    return donors
```
Comment (natasha-aleksandrova) [inline students/jared_mulholland/lesson_5/mail_room.py]: this function shouldn't have a return

File: students/jared_mulholland/lesson_5/mail_room.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        name = input("\nPlease Enter First and Last Name: ")
+
+    if name not in donors:
+        donors[name] = []
+        new_donation = input("Enter a Donation Amount: ")
+        new_donation = float(new_donation)
+        #append new_donation to donor list
+        donors[name].append(new_donation)            
+
+    elif name in donors:
```
Comment (natasha-aleksandrova) [inline students/jared_mulholland/lesson_5/mail_room.py]: this should be an else (not elif)

File: students/jared_mulholland/lesson_5/mail_room.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    -Finally, use string formatting to compose an email thanking the donor for their generous donation. Print the email to the terminal and return to the original prompt.
+    -It is fine (for now) to forget new donors once the script quits running.--
+"""
+
+def send_thankyou():
+    name = input("Please Enter First and Last Name: ")
+
+    while name == "list":
+        for donor in donors:
+            print(donor)
```
Comment (natasha-aleksandrova) [inline students/jared_mulholland/lesson_5/mail_room.py]: how could this be converted into a comprehension? (hint: `join`)

**Comment by jaredmulholland** [comment]
Thanks for the feedback. Can I add the changes to the Lesson 6 version of
mail room? Also, regarding lesson 6, I'm having a tough time figuring out
what to test. Testing functions that have an input (similar to the
exercises in the lesson) makes sense to me, but all of the functions in
mail room don't have an input. I know I'm missing something, but I was
hoping you could help me to figure out what I'm missing :).

Thanks,
jared

**Pull request merged**